src/select/multiselect/MultiSelect.js

```diff
@@ -58,7 +60,7 @@ class MultiSelect extends Component {
     }
 
     // Check the 'Select All' checkbox if all other values in the dropdown are selected
-    if (!containsVal && allowSelectAll && newValues.length === flatOptions.length - 1) {
+    if (!containsVal && allowSelectAll && newValues.length === unfilteredOptions.length - 1) {
```

Comment:
The main purpose of unfilteredOptions is so that when a multi-dropdown is filtered, none of the previously selected values that don't match the current filter criteria are lost.  Options will have the filtered options only and when trying to return the new value list, won't find anything previously selected.  For example say you have options like [1, 2, 3, 4, 5].  You open the dropdown , select 1, then close it.  Now type in the dropdown '3'.  Options will only contain the value 3.  Without the entire options list to search through, 1 isn't found in the list of options and is removed from the value array that's returned.  I'm open to suggestions to make that better though if you have thoughts.

Reply:
I think we can account for all other cases by handling `options` in the wrapper `getSelectValue`. In this case for this line, I think we can avoid using `unfilteredOptions` by adding a `isAllSelected` prop to `MultiSelect`. The default in `MultiSelect` would be `return newValues.length === flatOptions.length - 1`. But in `FilterSelect` it could be 
`
const { options } = this.props
return newValues.length === options.length
`

This could also be a useful override for other scenarios. Does that make sense?

Reply:
I did the first part with `getSelectValue`, that all makes sense now.  I'm confused as to how the `isAllSelected` option will help in the `onChange` function in `MultiSelect`.  We don't know if they're all selected until the `onChange` event occurs.  I was initially passing in a length property instead of the entire unfiltered property list, I could go back to something like that.